### DISCUSSION TITLE
How to use MarkerLabelFormatter?

### DISCUSSION THREAD

**Discussion by virtualpathum** [patrykandpatrick/vico#599] [Questions]
Hi, I need to customize the marker to add a prefix to each value and remove the decimal points. I tried creating the following custom label format but couldn't figure out how to assign this labelFormatter to the marker as there is no such property.

Then I tried MarkerComponent but still, there is no way.

I would appreciate it if someone could provide a sample working code to customize the marker.

Thanks in advance.

```
class CustomMarkerLabelFormatter : MarkerLabelFormatter {
    override fun getLabel(markedEntries: List<Marker.EntryModel>, chartValues: ChartValues): CharSequence {
        val builder = StringBuilder()
        for (entry in markedEntries) {
            builder.append("Prefix: ${entry.entry}\n") 
        }
        return builder.toString()
    }
}

 val marker = rememberMarker()
marker. labelFormatter = // There is no such property as labelFormatter 

MarkerComponent() How to use this?

```

Couldn't find any sample code in the given sample charts

**Comment by patrickmichalik** [comment]
Hello! I assume `rememberMarker` is the function from the sample app. `labelFormatter` is a field of `MarkerComponent`, but while `rememberMarker` uses `MarkerComponent`, its return type is `Marker`. You’ll have to either apply the `MarkerLabelFormatter` in `rememberMarker` or change the return type of `rememberMarker` to `MarkerComponent`.

**Reply by virtualpathum** [reply]
Hi @patrickmichalik Thanks,  Will try this and let you know

**Reply by patrickmichalik** [reply]
You’re welcome! Sounds good.
